extras/freenode_taxonomy_reader.rb

```diff
@@ -0,0 +1,59 @@
+require 'socket'
+require 'securerandom'
+require 'timeout'
+
+class FreenodeTaxonomyReader
+  NICKSERV = 'NickServ!NickServ@services.'
+  END_MESSGE_REGEX = /End of (.+) taxonomy|(.+) is not registered/
+
+  def initialize(
+        socket_provider: ->() { ssl_socket },
+        username_provider: ->() { "lobsters-#{SecureRandom.alphanumeric(8)}" }
+      )
+    @socket_provider = socket_provider
+    @nick = username_provider.call
+  end
+
+  def for_user(username)
+    _head, *taxonomy_lines = taxonomy_for(username)
+
+    taxonomy_lines.map do |line|
```

Comment:
I see how this splits apart the levels of abstraction - raw irc data from a proper hash. But given that we care about exactly one line, let's skip it and return/yield that one line if present and ignore everything else. Like the issue with someone throwing thousands of lines in their taxonomy, this also helps us if someone has a single really long line, we don't need to allocate state for it.

Reply:
This isn't resolved, `taxonomy_lines` returns an array when we only need it to return a string for the single username line.